### ISSUE TITLE
format.col_spec() fails with long expressions.

### ISSUE THREAD

**Issue by jdeboer** [tidyverse/readr#597]
I am defining a column as a factor, as follows.

```r
# Example 1:
readr::cols(a = readr::col_factor(
    levels = c("apple", "pear", "banana", "peach", "apricot", "orange", "plum"), ordered = TRUE)
)
```
Which produces the following error:

```
## Error in vapply(cols[[i]], deparse, character(1)) : 
##  values must be length 1,
## but FUN(X[[1]]) result is length 2 
```

However, the following two examples do not produce errors:
```r
# Example 2:
# Remove one of the levels to leave just six.
readr::cols(a = readr::col_factor(
   levels = c("apple", "pear", "banana", "peach", "apricot", "orange"), ordered = TRUE)
)
```
```r
# Example 3:
# Back to seven levels again, but with shorter labels.
readr::cols(a = readr::col_factor(
    levels = c("apple", "pear", "banana", "peach", "apri", "oran", "plum"), ordered = TRUE)
)
```

But with the third example, if I add just one extra character to any of the labels, e.g. 'apri' to 'apric', I get the error again:
```r
# Example 4:
readr::cols(a = readr::col_factor(
    levels = c("apple", "pear", "banana", "peach", "apric", "oran", "plum"), ordered = TRUE)
)
## Error in vapply(cols[[i]], deparse, character(1)) : 
##  values must be length 1,
## but FUN(X[[1]]) result is length 2 
```

After trying a number of combinations, I believe the error occurs once there are more than 32 cumulative characters used to define the labels of the levels:
```r
# From example 3:
nchar(paste(c("apple", "pear", "banana", "peach", "apri", "oran", "plum"), collapse = ""))
## [1] 32

# From example 4:
nchar(paste(c("apple", "pear", "banana", "peach", "apric", "oran", "plum"), collapse = ""))
## [1] 33
```

Below is a copy of the traceback:
```
## 7.
## vapply(cols[[i]], deparse, character(1)) 
## 6.
## FUN(X[[i]], ...) 
## 5.
## vapply(seq_along(cols), function(i) {
##     col_funs <- sub("^collector_", "col_", class(cols[[i]])[[1]])
##     args <- vapply(cols[[i]], deparse, character(1))
##     args <- paste(names(args), args, sep = " = ", collapse = ", ") ... 
## 4.
## format.col_spec(x, n = n, condense = condense, ...) 
## 3.
## cat(format.col_spec(x, n = n, condense = condense, ...)) 
## 2.
## print.col_spec(x) 
## 1.
## function (x, ...) 
## UseMethod("print")(x) 
```

I am using the current development version of `readr` available through github and R version 3.3.2 on Windows 10.

**Comment by jennybc** [comment]
You can *make* such a specification ... but you just can't print it at the moment.

``` r
x <- readr::cols(a = readr::col_factor(
  levels = c("apple", "pear", "banana", "peach", "apricot", "orange", "plum"), ordered = TRUE)
)
print(x)
#> Error in vapply(cols[[i]], deparse, character(1)): values must be length 1,
#>  but FUN(X[[1]]) result is length 2
```

I realize that is still a problem! But it means you can still use `col_factor()` in the meantime, as long you can suppress the printing of the col spec.

**Comment by jimhester** [comment]
As @jennybc said, this was an error in the format method for col_spec objects caused by deparse breaking long expressions into multiple lines. Fixed by #598.
